File: puzzles/swift/onboarding/onboarding.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
    public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

// game loop
while true {
    let enemy1 = readLine()! // name of enemy 1
    let distance1 = Int(readLine()!)! // distance to enemy 1
    let enemy2 = readLine()! // name of enemy 2
    let distance2 = Int(readLine()!)! // distance to enemy 2

    if distance1 < distance2 {
        print(enemy1)
    } else {
        print(enemy2)
    }
}

File: puzzles/python3/onboarding/onboarding.py
if __name__ == '__main__':
    while True:
        enemy_1 = input()  # name of enemy 1
        dist_1 = int(input())  # distance to enemy 1
        enemy_2 = input()  # name of enemy 2
        dist_2 = int(input())  # distance to enemy 2

        # Compare the distances of the two enemies
        if dist_1 < dist_2:
            # If enemy 1 is closer, shoot enemy 1
            print(enemy_1)
        else:
            # If enemy 2 is closer or at the same distance, shoot enemy 2
            print(enemy_2)
